### PR TITLE
esp32/i2c: Add macros to conform with other peripherals and fix typos

### PR DESCRIPTION
**Pull request by lucasssvaz**
## Summary

Added missing macros to the I2C files for the ESP32. Also fixed some typos.

## Impact

None

## Testing

Tested the macros by initializing devices that use I2C.

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/xtensa/src/esp32/esp32_i2c.h [hunk 1]
```diff
@@ -43,6 +43,14 @@ extern "C"
 #define EXTERN extern
 #endif
 
+#ifdef CONFIG_ESP32_I2C0
+  #define ESP32_I2C0 0
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32/esp32_i2c.h:47]:
```suggestion
#  define ESP32_I2C0 0
```

File: arch/xtensa/src/esp32/esp32_i2c.h [hunk 2]
```diff
@@ -43,6 +43,14 @@ extern "C"
...
+#endif /* CONFIG_ESP32_I2C0 */
+
+#ifdef CONFIG_ESP32_I2C1
+  #define ESP32_I2C1 1
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32/esp32_i2c.h:51]:
```suggestion
#  define ESP32_I2C1 1
```

**Review by pkarashchenko — COMMENTED** [review]
Please squash commits into 1

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Pull request merged**